py4cast/lightning.py

```diff
@@ -434,8 +440,18 @@ def _common_step(
                 # Only update the prev_states if we are not at the last step
                 if i < batch.num_pred_steps - 1 or k < num_inter_steps - 1:
                     # Update input states for next iteration: drop oldest, append new_state
-                    prev_states = torch.cat(
-                        [prev_states[:, 1:], new_state.unsqueeze(1)], dim=1
+                    prev_states = NamedTensor.new_like(
+                        torch.cat(
+                            [
+                                prev_states.index_select_dim(
+                                    "timestep",
+                                    range(1, prev_states.dim_size("timestep")),
+                                ),
+                                new_state.unsqueeze(batch.inputs.dim_index("timestep")),
+                            ],
+                            dim=batch.inputs.dim_index("timestep"),
+                        ),
+                        batch.inputs,
                     )
```

Comment:
this line is a bit scary ^^ could we split it in several explicit variables ?

Reply:
done